### PR TITLE
ui: Add Query Nagging for Sampling

### PR DESCRIPTION
**Pull request by zhangvi7**
### Context 

During query runs, if it is taking more than 10 seconds and the query could be sampled, we show a suggestion to users that they can cancel their current query and run a sampled version instead.

### Test Plan

naggingtest

### CODE REVIEW

File: querybook/webapp/components/QueryComposer/QueryComposer.tsx [hunk 1]
```diff
@@ -497,62 +497,80 @@ const QueryComposer: React.FC = () => {
         return getSelectedQuery(query, selectedRange);
     }, [query, queryEditorRef]);
 
+    const getSamplingTables = useCallback(
```
Comment (zhangvi7) [inline querybook/webapp/components/QueryComposer/QueryComposer.tsx:500]: @jczhong84 I added this to run If sampled_table is not provided while sampling is true and we'll use the TABLESAMPLE clause, if that looks fine to you

File: querybook/webapp/components/DataDocQueryCell/DataDocQueryCell.tsx [hunk 2]
```diff
@@ -409,7 +409,8 @@ class DataDocQueryCellComponent extends React.PureComponent<IProps, IState> {
     @bind
     public handleMetaChange<K extends keyof IDataQueryCellMeta>(
         field: K,
-        value: IDataQueryCellMeta[K]
+        value: IDataQueryCellMeta[K],
+        callback?: () => void
```
[the remark this answers is not in the document]
Comment (zhangvi7) [reply querybook/webapp/components/DataDocQueryCell/DataDocQueryCell.tsx]: removed with new style change

File: querybook/webapp/components/DataDocQueryExecutions/DataDocQueryExecutions.tsx [hunk 3]
```diff
@@ -24,136 +24,160 @@ interface IProps {
...
+            };
+
+            const selectedExecution = queryExecutions[selectedExecutionIndex];
+            const queryExecutionDOM = selectedExecution && (
+                <QueryExecution
+                    id={selectedExecution.id}
+                    docId={docId}
+                    key={selectedExecution.id}
+                    changeCellContext={changeCellContext}
+                    onSamplingInfoClick={onSamplingInfoClick}
```
Comment (zhangvi7) [inline querybook/webapp/components/DataDocQueryExecutions/DataDocQueryExecutions.tsx:156]: Only change in this file was passing 2 props here, rest at auto formatted changes

**Comment by zhangvi7** [comment]
- Removed rerun button + updated styling
- Updated test plan screenshot

File: querybook/webapp/components/DataDocQueryCell/DataDocQueryCell.tsx [hunk 4]
```diff
@@ -548,11 +548,14 @@ class DataDocQueryCellComponent extends React.PureComponent<IProps, IState> {
         );
 
         if (renderedQuery) {
+            const executionMetadata =
+                this.sampleRate > 0 ? { sample_rate: this.sampleRate } : null;
```
Comment (jczhong84) [inline querybook/webapp/components/DataDocQueryCell/DataDocQueryCell.tsx:552]: maybe have a getter function for the `executionMetadata`?

**Review by jczhong84 — APPROVED** [review]

**Pull request merged**